### PR TITLE
Put test/subtest name in backticks in s-c report

### LINKED ISSUE #4110: markup chars in test/subtest names break rendering of stability-checker report table
Some test/subtest names (e.g., many in http://w3c-test.org/html/infrastructure/urls/resolving-urls/query-encoding/windows-1251.html) have stuff like literal "`<table>`" in their names—that is, markup characters, and in particular "`<`".

When the stability checker output gets run through a markdown processor to render the report table it generates, those markup chars are treated as actual HTML markup—which breaks the rendering.

So we need to make `check_stability.py` output test/subtest names with backticks around them.

Also, some test/subtest names have line breaks in them (intentionally) as well as tabs and literal backticks. The newlines break the markdown formatting of the stability-checker report table (which expects all content for a table row to be on a single line). The backticks of course cause the markdown processor to think a span of code has ended or started, and the tab characters just show up as spaces.

So we need to translate all of those cases into something that works as expected with markdown.

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes #4110

### CODE REVIEW

File: check_stability.py [hunk 1]
```diff
@@ -26,6 +27,10 @@
 wptrunner = None
 
 
+reload(sys)
+sys.setdefaultencoding('utf8')
```
Comment (jgraham) [inline check_stability.py]: This is a hack that seems to be considered bad form (and isn't python 3 compatible). Let's explicitly encode things instead, as required.

File: check_stability.py [hunk 2]
```diff
@@ -402,6 +407,14 @@ def process_results(log, iterations):
     return results, inconsistent
 
 
+def markdown_adjust(s):
+    s = s.replace('\t', u'\u2409')
```
Comment (jgraham) [inline check_stability.py]: I'm not sure these first three aren't confusing. I guess to do this correctly we would need real table markup. In the interim replacing them with `\\t` etc. could be better.

**Comment by sideshowbarker** [comment]
Added https://github.com/w3c/web-platform-tests/pull/4111/commits/6bb279ee9c88a12309bc9573da1fdcd5c7f33678 in order to test the effects on rendering of the report table. For the resulting comment, see https://github.com/w3c/web-platform-tests/pull/4111#issuecomment-257553852.

https://github.com/w3c/web-platform-tests/pull/4111/commits/6bb279ee9c88a12309bc9573da1fdcd5c7f33678 needs to be dropped from the branch before merging to master.

File: check_stability.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 import os
 import re
 import stat
+import string
```
Comment (jgraham) [inline check_stability.py]: This seems to be unused.

Comment (jgraham) [inline check_stability.py]: This seems to be unused.

**Review by jgraham — APPROVED** [review]
lgtm with the unused import and test commit removed. Feel free to squash + merge with those changes.

**Pull request merged**